File: backend-v2/src/main/java/vn/io/vutiendat3601/shop/v2/order/OrderDto.java
package vn.io.vutiendat3601.shop.v2.order;

import java.math.BigDecimal;
import java.util.List;

public record OrderDto(
    String trackingNumber,
    OrderStatus status,
    Integer numberOfProducts,
    BigDecimal totalProductAmount,
    BigDecimal totalProductCouponAmount,
    BigDecimal totalProductFinalAmount,
    BigDecimal vatFeeAmount,
    BigDecimal shippingFeeAmount,
    BigDecimal shippingFeeCouponAmount,
    BigDecimal finalAmount,
    String shipingFeeCouponCode,
    String shippingAddressCode,
    String customerCode,
    List<OrderItemDto> items) {}

File: frontend-v2/src/app/domain/order/order-dto.ts
import { OrderItemDto } from './order-item-dto';

export class OrderDto {
  constructor(
    public trackingNumber: string,
    public status: string,
    public numberOfProducts: number,
    public totalProductAmount: number,
    public totalProductCouponAmount: number,
    public totalProductFinalAmount: number,
    public vatFeeAmount: number,
    public shippingFeeAmount: number,
    public shippingFeeCouponAmount: number,
    public finalAmount: number,
    public shipingFeeCouponCode: string,
    public shippingAddressCode: string,
    public customerCode: string,
    public items: OrderItemDto[]
  ) {}
}
